stackslib/src/core/mempool.rs

```diff
@@ -2206,10 +2209,58 @@ impl MemPoolDB {
         Ok(())
     }
 
-    /// Garbage-collect the mempool.  Remove transactions that have a given number of
-    /// confirmations.
+    /// Garbage-collect the mempool according to the behavior specified in `behavior`.
     pub fn garbage_collect(
-        tx: &mut MemPoolTx,
+        &mut self,
+        chain_height: u64,
+        behavior: &MempoolCollectionBehavior,
+        event_observer: Option<&dyn MemPoolEventDispatcher>,
+    ) -> Result<(), db_error> {
+        let tx = self.tx_begin()?;
+        match behavior {
+            MempoolCollectionBehavior::ByStacksHeight => {
+                let Some(min_height) = chain_height.checked_sub(MEMPOOL_MAX_TRANSACTION_AGE) else {
+                    return Ok(());
+                };
+                Self::garbage_collect_by_height(&tx, min_height, event_observer)?;
+            }
+            MempoolCollectionBehavior::ByReceiveTime => {
+                Self::garbage_collect_by_time(
+                    &tx,
+                    &MEMPOOL_NAKAMOTO_MAX_TRANSACTION_AGE,
+                    event_observer,
+                )?;
+            }
+        };
+        tx.commit()
+    }
+
+    /// Garbage-collect the mempool. Remove transactions that were accepted more than `age` ago.
+    /// The granularity of this check is in seconds.
+    pub fn garbage_collect_by_time(
+        tx: &MemPoolTx,
+        age: &Duration,
+        event_observer: Option<&dyn MemPoolEventDispatcher>,
+    ) -> Result<(), db_error> {
+        let threshold_time = get_epoch_time_secs().saturating_sub(age.as_secs());
+        let args: &[&dyn ToSql] = &[&u64_to_sql(threshold_time)?];
+        if let Some(event_observer) = event_observer {
+            let sql = "SELECT txid FROM mempool WHERE accept_time < ?1";
```

Comment:
Please make sure that the `accept_time` column is indexed.

Reply:
Yep, already was indexed: https://github.com/stacks-network/stacks-core/blob/develop/stackslib/src/core/mempool.rs#L786